src/osp/Resource/Resource.h

```diff
@@ -145,11 +145,14 @@ class DependRes
     }
 
     bool empty() const { return m_empty; }
+    operator bool() const { return !empty(); }
```

Comment:
Never make conversion operator to bool non-explicit.

Causes lots of screwups with people passing the type to a function and being surprised by conversions to bool.

Reply:
As in make it `explicit`? Requiring a `static_cast<bool>(resource)` is even more cumbersome than `(!resource.empty())` so I'll just remove the bool operator then.

Reply:
explicit bool operators continue working inside of if statements without needing to use either static cast, or !resource.empty().

Reply:
Oh, then I shouldn't have removed it -_- alright good to know

Reply:
This is the only real issue I have with this PR. Similarly, more explicit ifs like `a != 0` and `b != nulllptr` are more obvious and are preferred.